akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1197,7 +1197,10 @@ private[stream] object Collect {
             case Some(v) ⇒
               // #20217 the future is already here, avoid scheduling it on the dispatcher
               holder.setElem(v)
-              pushNextIfPossible()
+              v match {
```

Comment:
Maybe add another comment before this line:

```scala
// if `onComplete` is not used we need to check if we have to invoke the supervisor early because of a failure
```